src/common.speech/Transcription/ConversationServiceAdapter.ts

```diff
@@ -0,0 +1,648 @@
+import { ConversationTranslatorCommandTypes, ConversationTranslatorMessageTypes, IInternalParticipant } from "./ConversationTranslatorInterfaces";
+import { ConversationTranslatorRecognizer } from "./ConversationTranslatorRecognizer";
+import {
+    CommandResponsePayload,
+    IParticipantPayloadResponse,
+    IParticipantsListPayloadResponse,
+    ITranslationResponsePayload,
+    ParticipantPayloadResponse,
+    ParticipantsListPayloadResponse,
+    SpeechResponsePayload,
+    TextResponsePayload  } from "./ServiceMessages/Exports";
+
+/***
+ * The service adapter handles sending and receiving messages to the Conversation Translator websocket.
+ */
+export class ConversationServiceAdapter extends ServiceRecognizerBase {
+    private privConversationServiceConnector: ConversationTranslatorRecognizer;
+    private privConversationConnectionFactory: IConnectionFactory;
+    private privConversationAuthFetchEventId: string;
+    private privConversationAuthentication: IAuthentication;
+    private privConversationRequestSession: ConversationRequestSession;
+    private privConnectionConfigPromise: Promise<IConnection>;
+    private privConversationConnectionPromise: Promise<IConnection>;
+    private privConnectionLoop: Promise<IConnection>;
+    private terminateMessageLoop: boolean;
+    private privUtteranceId: string = "";
+    private privConversationIsDisposed: boolean;
+
+    public constructor(
+        authentication: IAuthentication,
+        connectionFactory: IConnectionFactory,
+        audioSource: IAudioSource,
+        recognizerConfig: RecognizerConfig,
+        conversationServiceConnector: ConversationTranslatorRecognizer) {
+
+        super(authentication, connectionFactory, audioSource, recognizerConfig, conversationServiceConnector);
+
+        this.privConversationServiceConnector = conversationServiceConnector;
+        this.privConversationAuthentication = authentication;
+        this.receiveMessageOverride = this.receiveConversationMessageOverride;
+        this.recognizeOverride = this.noOp;
+        this.connectImplOverride = this.conversationConnectImpl;
+        this.configConnectionOverride = this.configConnection;
+        this.fetchConnectionOverride = this.fetchConversationConnection;
+        this.disconnectOverride = this.privDisconnect;
+        this.privConversationRequestSession = new ConversationRequestSession(createNoDashGuid());
+        this.privConversationConnectionFactory = connectionFactory;
+        this.privConversationIsDisposed = false;
+    }
+
+    public isDisposed(): boolean {
+        return this.privConversationIsDisposed;
+    }
+
+    public dispose(reason?: string): void {
+        this.privConversationIsDisposed = true;
+        if (this.privConnectionConfigPromise) {
+            this.privConnectionConfigPromise.onSuccessContinueWith((connection: IConnection) => {
+                connection.dispose(reason);
+            });
+        }
+    }
```

Comment:
>} [](start = 4, length = 1)

Please add back the synchronous version of this method as well:

public sendMessage(message: string): void {
        this.fetchConversationConnection().onSuccessContinueWith((connection: IConnection) => {
            connection.send(new ConversationConnectionMessage(
                MessageType.Text,
                message));
        });
    }

Reply:
Added back.